src/pond/lib/timeseries.js

```diff
@@ -344,12 +344,15 @@ class TimeSeries {
     /**
      * Sets a new underlying collection for this TimeSeries.
      *
-     * @param {Collection}  collection The new collection
+     * @param {Collection}  collection       The new collection
+     * @param {boolean}     isChronological  Causes the chronological
+     *                                       order of the events to
+     *                                       not be checked
      *
-     * @return {TimeSeries}            A new TimeSeries
+     * @return {TimeSeries}                  A new TimeSeries
      */
-    setCollection(collection) {
-        if (!collection.isChronological()) {
+    setCollection(collection, isChronological = false) {
+        if (!isChronological && !collection.isChronological()) {
```

Comment:
It's modeled after this kind of underscore behavior
indexOf_.indexOf(array, value, [isSorted]) 
where isSorted flag tells the code that it can treat the array differently